tests/main/snap-quota-cpu/task.yaml

```diff
@@ -0,0 +1,144 @@
+summary: Functional test for cpu quota-related snap commands.
+
+details: |
+  Functional test for snap cpu quota group commands ensuring that they are 
+  effective in practice.
+
+# these systems do not support quota groups due to their old systemd versions,
+# we do have a check to do for these systems, but that is just to ensure that
+# we don't allow using quota groups, and that check is done in the snap-quota
+# spread instead
+systems:
+  - -ubuntu-14.04-*
+  - -centos-7-*
+  - -amazon-linux-2-*
+  - -ubuntu-16.04-*
+  - -ubuntu-core-16-*
+
+prepare: |
+  snap install test-snapd-stressd --edge --devmode
+  snap set system experimental.quota-groups=true
+  tests.cleanup defer snap unset system experimental.quota-groups
+
+restore: |
+  echo "Stopping the service"
+  snap stop test-snapd-stressd.stress-sc
+
+execute: |
+  echo "Starting service and verifying cpu usage"
+  snap start test-snapd-stressd.stress-sc
+
+  # helper function that has retry logic to allow for cpu usage to stabilize
+  function ensure_cpu_usage_higher_than() {
+    local service_name="$1"
+    local expected_usage="$(($2*100))"
+    local n
+    
+    n=0
+    until [ "$n" -ge 5 ]
+    do
+      # We use 'top' here to get the current CPU usage as top sorts by CPU usage, 
+      # so we assume that 'stress' is going to lie in the top as the odds of another
+      # process being in the top is close to 0.
+      usage=$(top -w256 -b -n 2 -d 0.2 | head | grep "$service_name" | awk '{print $9*100}')
```

Comment:
I'd suggest using `systemd-cgtop`, it's more reliable in that it considers all the processes in the cgroup, and you can specify the service you are interested in. Something like
```
usage=$(systemd-cgtop --iterations=2 "$service_NAME" | tail -1 | awk '{print $3*100}')
```

Reply:
Initially when calling this function, the service is not part of a specific slice, would this still work?